salt/utils/process.py

```diff
@@ -740,33 +753,51 @@ def __getstate__(self):
     def _setup_process_logging(self):
         salt.log.setup.setup_multiprocessing_logging(self.log_queue)
 
-    def run(self):
+    def _run(self):
         for method, args, kwargs in self._after_fork_methods:
             method(*args, **kwargs)
         try:
-            return super(MultiprocessingProcess, self).run()
```

Comment:
Hmm, tests pass , weirdly, but I actually have to still do some fixing, and find out why the tests passed.
We should be calling `self._origonal_run`

Reply:
Probably because tests are calling `run` and not `start`...

Reply:
Of course, the test suite didn't even start.
Will still investigate the unit tests passing.

Reply:
Of course we can't call `.start()`, the mocks will get inherited by the new process, on Linux, but we can no longer assert against them, at least like we usually do.